src/cmd/build.rs

```diff
@@ -183,14 +199,18 @@ fn exec_cargo_for_wasm_target(
 
     let cargo_build = |manifest_path: &ManifestPath| {
         let target_dir = &crate_metadata.target_directory;
-        let args = [
+        let target_dir = format!("--target-dir={}", target_dir.to_string_lossy());
+        let mut args = vec![
             "--target=wasm32-unknown-unknown",
             "-Zbuild-std",
             "-Zbuild-std-features=panic_immediate_abort",
             "--no-default-features",
             "--release",
```

Comment:
Shouldn't this only be added depending on the `BuildMode` now?

Reply:
I think we'd still want to build the wasm in release mode for optimal size, after all these blobs are still deployed to the chain. That said, I've not tested what the difference is building without `--release`.